shared/src/ledger/protocol/transactions/ethereum_events/mod.rs

```diff
@@ -157,17 +157,9 @@ where
     // is a less arbitrary way to do this
     let (exists_in_storage, _) = storage.has_key(&eth_msg_keys.seen())?;
 
-    let mut seen_by = Votes::default();
-    for (address, block_height) in update.seen_by.into_iter() {
-        // TODO: more deterministic deduplication
-        if let Some(present) = seen_by.insert(address, block_height) {
-            tracing::warn!(?present, "Duplicate vote in digest");
-        }
-    }
```

Comment:
This block actually wasn't necessary since we switched `seen_by` to be a `BTreeMap<Address, BlockHeight>` already

Reply:
I think this is also present in the validator set update code. could you update it in this PR, as well? I basically just copied and pasted your eth events code and modified it a bit

Reply:
Leaving this as is for now but we should revisit later the `ValidatorSetUpdateVextDigest` as we should always be able to construct it without duplicates anyway

Reply:
Will quickly update this TODO